src/tengbe.py

```diff
@@ -276,36 +276,29 @@ def read_counters(self):
                 results[reg] = self.parent.memory_devices[reg].read()
         return results
 
-    def tx_okay(self,wait_time=1):
+    def tx_okay(self, wait_time=1):
         """
         Is this gbe block okay?
         :return: True/False
         """
-        try:
-            result0 = self.read_tx_counters()
-            time.sleep(wait_time) # what number is best?
-            result1 = self.read_tx_counters()
-            # if debug register are implemented
-            # check if errors are not changing
-            assert (result0[self.name+'_txfullctr']['data']['reg'] ==
-                    result1[self.name+'_txfullctr']['data']['reg'])
-            assert (result0[self.name+'_txofctr']['data']['reg'] ==
-                    result1[self.name+'_txofctr']['data']['reg'])
-            assert (result0[self.name+'_txerrctr']['data']['reg'] ==
-                    result1[self.name+'_txerrctr']['data']['reg'])
-            # check increment
-            assert (result1[self.name+'_txctr']['data']['reg'] -
-                    result0[self.name+'_txctr']['data']['reg'] > 0)
-            assert (result1[self.name+'_txvldctr']['data']['reg'] -
-                    result0[self.name+'_txvldctr']['data']['reg'] > 0)
-        except KeyError:
-            raise RuntimeError('Gbe block {} does not have the necessary '
-                               'registers to check TX status.'.format(self.name))
-        except:
-            LOGGER.info('Gbe block %s tx_okay() - FALSE.' % self.name)
+        result0 = self.read_tx_counters()
+        time.sleep(wait_time)  # what number is best?
+        result1 = self.read_tx_counters()
+        if ((result0[self.name+'_txfullctr']['data']['reg'] ==
```

Comment:
Okay, so next thing: the gbe core can be compiled with only SOME of the debug registers in it. The casperfpga library handles this and the read_tx_counters() function will only read the ones that are there. You might get some, you might get none.

For your purposes, you must have _txctr incrementing and _txerrctr not incrementing. The registers must be there. The should indicate tx with no failures.

For the others, issue a warning if they are not there, but fail hard if they are there and indicate failures.